### PR TITLE
LPD-23020 - Screen is Not Properly Read in Required Field Error Messages with JAWS Screen Reader 

### PR DESCRIPTION
**Pull request by ilzamcmed**
Jira issue: https://liferay.atlassian.net/browse/LPD-23020

**Error**: After pressing `tabKey`, the name of the field you land on is ignored and we can only hear the error message.

**Expected Result**: After pressing `tabKey`, the name of the field you land on and the error message should be read out if the previous field wasn't filled.

Evidence:

https://github.com/liferay-forms/liferay-portal/assets/47724428/c56bfb9d-4a7c-4429-a51f-ca7b5393f329

### CODE REVIEW

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:sf

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/FieldFeedback.tsx [hunk 1]
```diff
@@ -15,30 +15,25 @@ export function FieldFeedback({
...
 	...otherProps
 }: IProps) {
-	if (!errorMessage && !helpMessage && !warningMessage) {
-		return null;
-	}
+	const message =
+		warningMessage && !errorMessage ? warningMessage : errorMessage;
+
+	const symbol =
+		warningMessage && !errorMessage ? 'warning-full' : 'exclamation-full';
```
Comment (ethib137) [inline modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/FieldFeedback.tsx:22]: In these two ternary statements you are doing the exact same check. In this situation it would be better to only make the check once and reuse it. This can be done with an `if` block or by declaring a variable and using that variable in the ternary statements: `const showWarning = warningMessage && !errorMessage;`

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/FieldFeedback.tsx [hunk 2]
```diff
@@ -15,30 +15,25 @@ export function FieldFeedback({
...
-				</ClayForm.FeedbackItem>
-			)}
+			<ClayForm.FeedbackItem
+				aria-live="assertive"
+				id={`${name}_fieldError`}
+			>
+				{message && <ClayForm.FeedbackIndicator symbol={symbol} />}
+
+				{message}
+			</ClayForm.FeedbackItem>
```
Comment (ethib137) [inline modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/FieldFeedback.tsx:36]: There is a pretty large change to the logic here. This change causes the `<ClayForm.FeedbackGroup>` and the `<ClayForm.FeedbackItem>` to always be rendered when previously they were only rendered when if there was a message. What is the reason for this change? Is there any unintended spacing changes that occurs as a result?

**Review by ethib137 — CHANGES_REQUESTED** [review]
Hey @ilzamcmed thanks for working on this. Can you make sure to provide an explanation for the changes you are making in the PR? That way we can more easily determine what the intent is. Thank you!

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/FieldFeedback.tsx (hunk 2 above)
Comment (ilzamcmed) [reply modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/FieldFeedback.tsx:36]: Thanks @ethib137!
The reason I made this change, was that Screen Reader didn't recognize the `id` on the feedback because it was not rendered yet, and because of that, it did not read the error. 
I haven't noticed any change in spacing but I'll check!

Comment (ilzamcmed) [reply modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/FieldFeedback.tsx:36]: There is no space until the message is displayed.

Screenshot_20240415_085208

File: modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/FieldFeedback.tsx (hunk 1 above)
Comment (ilzamcmed) [reply modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/FieldFeedback.tsx]: Fixed!

**Comment by igor-franca** [comment]
ci:test:sf

**Review by ethib137 — APPROVED** [review]
LGTM

**Review by igor-franca — APPROVED** [review]

**Comment by igor-franca** [comment]
ci:test:data-engine

**Comment by igor-franca** [comment]
Hi @ilzamcmed, could you run the test `DEFieldReference#CanNotBeTheSameForTwoOptionsOfFields` in your local environment and try to check this error:

`java.lang.Exception: Actual text "" does not match pattern "(?iu)This reference is already being used\. Try a different one\." at "xpath=(//div[contains(@class,'ddm-option-entry')])[2]//div[contains(@class,'form-feedback-item')]"`

I managed to investigate the other tests that failed, and only this one caught my attention, as it was a unique failure and was related to feedback from a field.

Thanks in advance

**Comment by ilzamcmed** [comment]
Hey @igor-franca, I ran the test locally on this branch (LPD-23020) and this particular test passed successfully.

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:data-engine

**Comment by igor-franca** [comment]
ci:test:relevant

**Comment by ilzamcmed** [comment]
Hi  @igor-franca, I was finally able to reproduce the poshi error and I am working on the fix.
Thanks

**Comment by ilzamcmed** [comment]
ci:test:sf

**Comment by ilzamcmed** [comment]
ci:test:relevant

**Comment by ilzamcmed** [comment]
ci:test:data-engine

**Comment by igor-franca** [comment]
After checking the [comparison](https://testray.liferay.com/home/-/testray/runs/compare?formDate=1713480333925&testrayRunIdA=2561842671&testrayRunIdB=2561940816) between the tests performed in the PR and the routine result, I was unable to identify any test that showed an error related to the changes proposed in the PR.

**Comment by igor-franca** [comment]
ci:forward:force
